src/HttpCache/SouinPurger.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\HttpCache;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Purges Souin.
+ *
+ * @author Sylvain Combraque <darkweak@protonmail.com>
+ *
+ * @experimental
+ */
+class SouinPurger implements PurgerInterface
+{
+    private const MAX_HEADER_SIZE_PER_BATCH = 1500;
+    private const SEPARATOR = ', ';
+    private const SEPARATOR_LENGTH = 2;
+    private const HEADER = 'Surrogate-Key';
+
+    /**
+     * @var HttpClientInterface[]
+     */
+    private $clients;
+
+    /**
+     * The dsn to access to the remote purger instance.
+     *
+     * @var string
+     */
+    private $dsn;
+
+    /**
+     * The URL path to access to the remote purger instance.
+     *
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @param HttpClientInterface[] $clients
+     */
+    public function __construct(
+        array $clients,
```

Comment:
Could we use property Promotion and make them readonly here?
Apip 3.0 requires PHP 8.1